pkg/docker/config/config.go

```diff
@@ -739,13 +752,27 @@ func stripScheme(url string) string {
 	return stripped
 }
 
+// normalizeRegistry converts the provided registry to contain only the hostname.
 func normalizeRegistry(registry string) string {
 	normalized := convertToHostname(registry)
-	switch normalized {
+	return convertDockerHost(normalized)
+}
+
+// stripRegistry removes known prefixes as well as suffixes from the registry.
+func stripRegistry(registry string) string {
+	stripped := stripScheme(registry)
+	stripped = strings.TrimSuffix(stripped, "/v1/")
```

Comment:
AFAICS this matches `quay.io/v1/` and `quay.io/foo/v1/`, differentiating from scoped keys `quay.io/v1` and `quay.io/foo/v1` only using the trailing slash.

And historically `normalizeRegistry` has matched arbitrary URLs, including `https://example.com/an/ignored/comment`, so relying on a single trailing slash feels rather brittle.

Why not normalize only keys that start with a scheme`://` (and then probably ignore the path part completely)?

Reply:
Alright, switched to the conversion approach which makes the whole function obsolete.